src/commands/force/mdapi/beta/retrieve.ts

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import * as fs from 'fs';
+import { resolve } from 'path';
+import { flags, FlagsConfig } from '@salesforce/command';
+import { Messages, SfdxError, SfdxProject } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { MetadataApiRetrieve, RequestStatus, RetrieveResult } from '@salesforce/source-deploy-retrieve';
+import { Optional } from '@salesforce/ts-types';
+import { SourceCommand } from '../../../../sourceCommand';
+import { Stash } from '../../../../stash';
+import { ComponentSetBuilder } from '../../../../componentSetBuilder';
+import { FsError } from '../../../../types';
+import {
+  RetrieveCommandResult,
+  RetrieveCommandAsyncResult,
+  RetrieveResultFormatter,
+} from '../../../../formatters/mdapi/retrieveResultFormatter';
+
+interface EnsureFlagOptions {
+  flagName: string;
+  path: string;
+  type: 'dir' | 'file' | 'any';
+  throwOnENOENT?: boolean;
+}
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-source', 'md.retrieve');
+const spinnerMessages = Messages.loadMessages('@salesforce/plugin-source', 'spinner');
+
+export class Retrieve extends SourceCommand {
+  public static readonly description = messages.getMessage('retrieveCmd.description');
+  public static readonly examples = messages.getMessage('retrieveCmd.examples').split(os.EOL);
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    retrievetargetdir: flags.directory({
+      char: 'r',
+      description: messages.getMessage('flags.retrievetargetdir'),
+      longDescription: messages.getMessage('flagsLong.retrievetargetdir'),
+      required: true,
+    }),
+    unpackaged: flags.filepath({
+      char: 'k',
+      description: messages.getMessage('flags.unpackaged'),
+      longDescription: messages.getMessage('flagsLong.unpackaged'),
+      exclusive: ['sourcedir', 'packagenames'],
+    }),
+    sourcedir: flags.directory({
+      char: 'd',
+      description: messages.getMessage('flags.sourcedir'),
+      longDescription: messages.getMessage('flagsLong.sourcedir'),
+      exclusive: ['unpackaged', 'packagenames'],
+    }),
+    packagenames: flags.array({
+      char: 'p',
+      description: messages.getMessage('flags.packagenames'),
+      longDescription: messages.getMessage('flagsLong.packagenames'),
+      exclusive: ['sourcedir', 'unpackaged'],
+    }),
+    singlepackage: flags.boolean({
+      char: 's',
+      description: messages.getMessage('flags.singlepackage'),
+      longDescription: messages.getMessage('flagsLong.singlepackage'),
+    }),
+    zipfilename: flags.string({
+      char: 'n',
```

Comment:
```suggestion
      char: 'f',
```

Reply:
We use "n" for names, but "f" for filenames.  At least, according to this: https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_style_parameter_names.htm